### PR TITLE
feat(config): Support setting `SELENIUM_PROMISE_MANAGER` flag via the…

### PR DESCRIPTION
**Pull request by sjelin**
… config

Closes https://github.com/angular/protractor/issues/3691

### CODE REVIEW

**Comment by sjelin** [comment]
CircleCI: :white_check_mark:

**Comment by sjelin** [comment]
Travis: :white_check_mark:

File: lib/config.ts [hunk 1]
```diff
@@ -601,6 +601,29 @@ export interface Config {
...
+   *
+   * WebDriverJS (and by extention, Protractor) uses a Control Flow to manage the order in which
+   * commands are executed and promises are resolved (see docs/control-flow.md for details).
+   * However, as syntax like `async`/`await` are being introduced, WebDriverJS has decided to
+   * deprecate the control flow, and have users manage the asynchronous activity themselves
+   * (details here: https://github.com/SeleniumHQ/selenium/issues/2969).
+   *
+   * At the moment, by default, the WebDriver control flow is still enabled.  You can disable it by
+   * setting the environment variable `SELENIUM_PROMISE_MANAGER` to `0`.  In about one year, the
+   * control flow will be disabled by default, and you will be able to re-enable it by setting
```
Comment (juliemr) [inline lib/config.ts]: Instead of 'in about one year', maybe "in a webdriver release in Q4 2018"

File: lib/config.ts [hunk 2]
```diff
@@ -601,6 +601,29 @@ export interface Config {
...
+   * `SELENIUM_PROMISE_MANAGER` to `1`.  In about two years, the control flow will be removed for
```
Comment (juliemr) [inline lib/config.ts]: Instead of "in about two year", "at a later point"

File: lib/config.ts [hunk 3]
```diff
@@ -601,6 +601,29 @@ export interface Config {
...
+   * good.
+   *
+   * If you don't like managing environment variables, you can set this option in your config file,
+   * and Protractor will handle enabling/disabling the control flow for you.  Setting this option
+   * will override the `SELENIUM_PROMISE_MANAGER` environment variable.
```
Comment (juliemr) [inline lib/config.ts]: Instead say "Setting this option is higher priority than the `SELENIUM_PROMISE_MANAGER` environment variable".

File: spec/ts/noCFSmokeConf.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import {Config} from '../..';
+const env = require('../environment.js');
+
+
+// The main suite of Protractor tests.
```
Comment (juliemr) [inline spec/ts/noCFSmokeConf.ts]: Please fix this comment.

File: spec/ts/noCF/smoke_spec.ts [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+        element(by.binding('item.reusedBinding'));
+
+    await browser.get('index.html#/conflict');
+
+    await expect(reused.getText()).toEqual('Inner: inner');
+  });
+
+  it('should allow handling errors', async function() {
+    await browser.get('index.html#/form');
+    await $('.nopenopenope').getText().then(async function(/* string */) {
```
Comment (juliemr) [inline spec/ts/noCF/smoke_spec.ts]: Would it make sense for this test to instead use try/catch (that works with async, yeah?)

**Comment by juliemr** [comment]
Whoo! A few comments, mostly on wording.

File: spec/ts/noCF/smoke_spec.ts (hunk 5 above)
Comment (sjelin) [inline spec/ts/noCF/smoke_spec.ts]: OMG YOU'RE RIGHT

**Comment by sjelin** [comment]
@juliemr all comments addressed

**Pull request merged**